mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/annotation/ExclusiveOptionalsTypeAnnotation.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.declaration.type.annotation;
+
+import org.mule.metadata.api.annotation.TypeAnnotation;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.runtime.extension.api.annotation.param.ExclusiveOptionals;
+
+import java.util.Set;
+
+/**
+ *  * A {@link TypeAnnotation} used to enrich an {@link ObjectType} by specifying
+ *  that the contained parameters have an exclusive relation as it is specified in {@link ExclusiveOptionals}
+ * <p>
+ * This class is immutable.
+ *
+ * @since 1.0
+ */
+public class ExclusiveOptionalsTypeAnnotation implements TypeAnnotation {
```

Comment:
this affects also the DTO model, talk to @gsfernandes

Reply:
@gsfernandes said 👍